src/app/views/contacts/contact-details/contact-details.component.ts

```diff
@@ -19,8 +19,8 @@ export class ContactDetailsComponent implements OnInit, OnDestroy {
 
   constructor(
       private activatedRoute: ActivatedRoute,
-      private router: Router,
-      private contactsFacade: ContactsStoreFacade,
+      public router: Router,
```

Comment:
Keep fields private, use dependency resolvers to get injected values: 
https://codecraft.tv/courses/angular/unit-testing/dependency-injection/#_resolving_via_the_component_injector

Reply:
@avatsaev thanks for sharing a helpful knowledge, now I am updating the PR.